apis/eventing/v1alpha1/cloudeventsource_types.go

```diff
@@ -22,6 +22,17 @@ import (
 	v1alpha1 "github.com/kedacore/keda/v2/apis/keda/v1alpha1"
 )
 
+// +kubebuilder:object:generate=false
+type CloudEventSourceInterface interface {
+	GetKind() string
+	GetName() string
+	GetNamespace() string
+	GetSpec() CloudEventSourceSpec
+	GetStatus() CloudEventSourceStatus
+	GetGeneration() int64
+	GenerateIdentifier() string
+}
```

Comment:
this can embed `client.Object` from controller-runtime to make a good amount of code simpler
```suggestion
type CloudEventSourceInterface interface {
    client.Object
	GetSpec() CloudEventSourceSpec
	GetStatus() CloudEventSourceStatus
	GenerateIdentifier() string
}
```